File: backend/src/main/java/co/gerard/grupoestudio/modelo/estudiante/Estudiante.java
package co.gerard.grupoestudio.modelo.estudiante;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "estudiantes")
public class Estudiante extends PanacheEntityBase {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "est_id")
    public UUID estId;

    @Column(name = "est_nombres", nullable = false, length = 50)
    public String estNombres;

    @Column(name = "est_apellidos", nullable = false, length = 50)
    public String estApellidos;

    @Column(name = "est_email", length = 100)
    public String estEmail;

    @Column(name = "est_creacion")
    public LocalDateTime estCreacion;

    // Método estático tipado para no tener que castear en el Resource
    public static Estudiante findById(UUID id) {
        return find("estId", id).firstResult();
    }
}

File: frontend/lib/modelos/estudiante.dart
class Estudiante {
  final String? estId;
  final String estNombres;
  final String estApellidos;
  final String? estEmail;

  Estudiante({
    this.estId,
    required this.estNombres,
    required this.estApellidos,
    this.estEmail,
  });

  factory Estudiante.fromJson(Map<String, dynamic> json) => Estudiante(
        estId: json['estId'],
        estNombres: json['estNombres'] ?? '',
        estApellidos: json['estApellidos'] ?? '',
        estEmail: json['estEmail'],
      );

  String get nombreCompleto => '$estNombres $estApellidos';
}